aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/service/StorageService.java

```diff
@@ -116,9 +115,14 @@ TransferObserver uploadInputStream(@NonNull String transferId,
      *
      * @param path path inside storage to inspect for list of items
      * @param prefix path appended to S3 keys
+     * @param subpathStrategy SubpathStrategy to include/exclude sub-paths
      * @return A list of parsed items present inside given path
      */
-    List<StorageItem> listFiles(@NonNull String path, @NonNull String prefix);
+    StorageListResult listFiles(
```

Comment:
This should be a new function - we can't add parameters to this function since it's public

Reply:
Updated to new function.